drivers/adc/adm1177/adm1177.c

```diff
@@ -0,0 +1,341 @@
+ * 			or ADM1177_REG_CONTROL will be used.
+ * 			In case of a write command byte operation reg_address
+ * 			will be set to 0.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_write(struct adm1177_dev *device, uint8_t cmd, uint8_t reg_address)
+{
+	if (reg_address & ADM1177_EXT_CMD_MASK) {
+		uint8_t data[2] = {reg_address, cmd};
+
+		return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+	}
+	if (cmd == ADM1177_CMD_I_CONT || cmd == ADM1177_CMD_I_ONCE) {
+		device->last_command = ADM1177_CURRENT_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	if (cmd == ADM1177_CMD_V_CONT || cmd == ADM1177_CMD_V_ONCE) {
+		device->last_command = ADM1177_VOLTAGE_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	if ((cmd == (ADM1177_CMD_I_CONT | ADM1177_CMD_V_CONT)) ||
+	    (cmd == (ADM1177_CMD_I_ONCE | ADM1177_CMD_V_ONCE))) {
+		device->last_command = ADM1177_VOLTAGE_AND_CURRENT_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	return -EINVAL;
+}
+
+/**************************************************************************//**
+ * @brief - Reads the status byte from the ADC if requested and transmits it
+ * 	    through the status_byte pointer.
+ *
+ * @param device - The device structure.
+ *
+ * @param status_byte - The pointer through which the status byte is
+ * 			transmitted.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read_status(struct adm1177_dev *device, uint8_t *status_byte)
+{
+	int ret;
+	if (device->last_command == ADM1177_STATUS_EN) {
+		if (status_byte == NULL)
+			return -EINVAL;
+
+		ret = no_os_i2c_read(device->i2c_desc, status_byte, 1, 1);
+		if (ret)
+			return ret;
+
+		return 0;
+	}
+	return -EINVAL;
+}
+
+/**************************************************************************//**
+ * @brief - Gets the voltage/current readback from the ADM1177s I2C ADC,
+ *	    if the device is initialized right and transmits the requested
+ *	    values.
+ *
+ * @param device - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @param conv_current - The variable through which the current read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read_conv(struct adm1177_dev *device,
+		      uint16_t* conv_voltage,
+		      uint16_t* conv_current)
+{
+	int ret;
+
+	uint8_t data[3];
+
+	switch (device->last_command) {
+	case ADM1177_CURRENT_EN:
+		if (conv_current == NULL)
+			return -EINVAL;
+
+		no_os_udelay(150);
```

Comment:
i suppose this is the ADC Conversion time from the datasheet specs. Why duplicating it 3 times instead of moving it before the switch case? Also, a comment would be nice for this "magic" number. See: https://github.com/analogdevicesinc/no-OS/blob/c9e00645575f1a02d46dfbb72cbf283324f6d265/drivers/adc-dac/ad74416h/ad74416h.c#L872-L873

Reply:
Will handle it.